x-pack/plugins/uptime/server/lib/alerts/tls.ts

```diff
@@ -128,27 +109,39 @@ export const tlsAlertFactory: UptimeAlertTypeFactory<ActionGroupIds> = (_server,
 
       const foundCerts = total > 0;
 
+      const uniqueCerts = certs.reduce<Cert[]>((uniqueCertArray, currentCert) => {
+        if (!uniqueCertArray.find((cert) => cert.common_name === currentCert.common_name)) {
+          uniqueCertArray.push(currentCert);
+        }
+
+        return uniqueCertArray;
+      }, []);
```

Comment:
i don't think we need this reduce, if you check getCerts function, es query is a collapse on `tls.server.hash.sha256` which means it will only return unique certificates. common_name actually can be same, `tls.server.hash.sha256 `will always be different.

Reply:
Thanks. I will use `sha256` for the id then, as using `common_name` for the individual alert instance id was causing duplicate scheduling issues.

Reply:
i would say keep common name at start of instance id and sha256 after that. sha256 isn't very helpful in identification.

Reply:
maybe append common name + index or something?

Reply:
As discussed offline the id will be common name + issuer + sha256

Reply:
are we keeping this reduce? i think we don't need this.

Reply:
@shahzad31 , you're right, my mistake.